alodataset/coco_base_dataset.py

```diff
@@ -78,10 +78,17 @@ def __init__(
             return
         else:
             assert img_folder is not None, "When sample = False, img_folder must be given."
-            assert ann_file is not None, "When sample = False, ann_file must be given."
+            assert ann_file is not None or  "test"  in img_folder, "When sample = False and the test split is not used, ann_file must be given."
+
 
         # Create properties
         self.img_folder = os.path.join(self.dataset_dir, img_folder)
+
+        if "test" in img_folder:
+            #get a list of  indices that don't rely on the annotation file
+            self.items= list(sorted([int(Path(os.path.join(self.img_folder, f)).stem) for f in os.listdir(self.img_folder) if os.path.isfile(os.path.join(self.img_folder, f))]))
```

Comment:
is there any reason for sorting this list ? it might be expensive, specially with the length of the variables you have (paths)

Reply:
I actually commited this file by mistake, it was already merged in #247